src/components/Header/Header.tsx

```diff
@@ -0,0 +1,73 @@
+import ArrowBackIosIcon from '@material-ui/icons/ArrowBackIos'
+import { useHistory } from 'react-router'
+import styled from 'styled-components'
+import Button from '../common/Button/Button'
+
+interface Props {
+  withBack?: Boolean
+  children: string
+}
+
+function Header({ withBack = false, children }: Props): JSX.Element {
+  const { goBack } = useHistory()
+
+  return (
+    <Container>
+      {withBack && (
+        <StrippedButton onClick={goBack}>
+          <ArrowBackIosIcon /> back
+        </StrippedButton>
+      )}
+      <StyledTitle>{children}</StyledTitle>
+    </Container>
+  )
+}
+
+export default Header
+
+const Container = styled.header`
+  display: flex;
+  justify-content: center;
+  position: relative;
+
+  &::after {
+    background-image: linear-gradient(#5c5c5c33, transparent);
+    content: '';
+    height: 4px;
+    left: 0;
+    position: absolute;
+    top: 100%;
+    width: 100%;
+  }
+`
+const StrippedButton = styled(Button)`
+  align-items: center;
+  background-color: transparent;
+  border-radius: 0;
+  border: none;
+  color: var(--color-text);
+  display: flex;
```

Comment:
there is nothing wrong by sorting the styles ascending, but for me its always wired while reading, that thinks that depends on each other are spread apart like align-items and display flex or one above  left, position, top. for me its always feels like it should be in the order position, top, left. Just to consider about.